In story viewer, loading a section updates Backbone's router location wi...

patternleaf opened the pull request:
...thout creating a new entry in the browser's history. Addresses #487.

patternleaf committed:
In story viewer, loading a section updates Backbone's router location without creating a new entry in the browser's history. Addresses #487.

ghing commented:
This doesn't seem to fix the issue.  I've updated the issue to make the incorrect behavior a little bit more clear.

ghing commented:
@patternleaf, was this working for you in both Chrome and Firefox?  My quick testing found that it only behaved correctly in Firefox.

patternleaf committed:
Create "iframe" mode for viewer (#487).

Allow for story viewer request with ?context=iframe. In this mode,
story navigation doesn't write to the browser history.

Also, take out previous "fix" for #487. The replace mode for the
router wasn't behaving consistently across browsers and other
attempts to make interaction with browser history sane weren't
working.

ghing commented:
This looks pretty good and held up to some quick testing.  I'm going to merge and then do a little clean-up post-merge.
